knowledge_runtime/knowledge_runtime/services/duckdb_generator.py

```
@@ -0,0 +1,709 @@
+    source_file_size: int = 0
+
+
+class DuckDBGenerator:
+    """Generates .duckdb files from Excel/CSV binary data.
+
+    Handles multi-sheet Excel files, CSV/TSV imports, and legacy .xls format.
+    Produces a DuckDB database file with SUMMARIZE metadata and sample data.
+    """
+
+    # Supported file extensions
+    SUPPORTED_EXTENSIONS = {".xlsx", ".xls", ".csv", ".tsv"}
```

Ruff (0.15.12)

[warning] Mutable default value for class attribute

(RUF012)